apps/daimo-mobile/src/view/screen/DepositScreen.tsx

```diff
@@ -0,0 +1,212 @@
+      <ScrollView>
+        <CoverGraphic type="deposit" />
+        <Spacer h={8} />
+        <DepositList account={account} />
+        <Spacer h={16} />
+        <WithdrawList />
+      </ScrollView>
+    </View>
+  );
+}
+
+function DepositList({ account }: { account: Account }) {
+  const isTestnet = chainConfig.chainL2.testnet;
+
+  const [started, setStarted] = useState(false);
+
+  const openExchange = (url: string) => {
+    Linking.openURL(url);
+    setStarted(true);
+  };
+
+  const dispatcher = useContext(DispatcherContext);
+
+  const openAddressDeposit = () => {
+    dispatcher.dispatch({ name: "depositAddress" });
+  };
+
+  const defaultLogo = `${daimoDomainAddress}/assets/deposit/deposit-wallet.png`;
+
+  const options: OptionRowProps[] = [
+    {
+      cta: "Deposit to address",
+      title: "Send USDC to your address",
+      logo: defaultLogo,
+      onClick: openAddressDeposit,
+    },
+  ];
+
+  if (!isTestnet) {
+    options.unshift(
+      ...account.recommendedExchanges.map((rec) => ({
+        title: rec.title || "Loading...",
+        cta: rec.cta,
+        logo: rec.logo || defaultLogo,
+        isExternal: true,
+        onClick: () => openExchange(rec.url),
+      }))
+    );
+  }
+
+  return (
+    <View style={{ marginHorizontal: 16 }}>
+      <DaimoText style={{ ...ss.text.body, color: color.gray3 }}>
+        Deposit
+      </DaimoText>
+      {started && (
+        <>
+          <Spacer h={16} />
+          <InfoBox
+            icon="check"
+            title="Deposit initiated"
+            subtitle="Complete in browser, then funds should arrive in a few minutes."
+          />
+        </>
+      )}
+      <Spacer h={16} />
+      {options.map((option) => (
+        <OptionRow key={option.cta} {...option} />
+      ))}
+    </View>
+  );
+}
+
+function WithdrawList() {
+  const dispatcher = useContext(DispatcherContext);
+
+  const openAddressWithdraw = () => {
+    dispatcher.dispatch({ name: "withdrawInstructions" });
+  };
+
+  const defaultLogo = `${daimoDomainAddress}/assets/deposit/withdraw-wallet.png`;
+
+  return (
+    <View style={{ marginHorizontal: 16 }}>
+      <DaimoText style={{ ...ss.text.body, color: color.gray3 }}>
+        Withdraw
+      </DaimoText>
+      <Spacer h={16} />
+      <OptionRow
+        cta="Withdraw"
+        title="Withdraw to any wallet or exchange"
+        logo={defaultLogo}
+        onClick={openAddressWithdraw}
+      />
+    </View>
+  );
+}
+
+type OptionRowProps = {
+  title?: string;
+  cta: string;
+  logo: string;
+  isExternal?: boolean;
+  onClick: () => void;
+};
+
+function OptionRow({ title, cta, logo, isExternal, onClick }: OptionRowProps) {
+  const width = useWindowDimensions().width;
+
+  return (
+    <TouchableHighlight
+      onPress={onClick}
+      {...touchHighlightUnderlay.subtle}
+      style={{ marginHorizontal: -16 }}
+    >
+      <View style={styles.optionRow}>
+        <View style={{ ...styles.optionRowLeft, maxWidth: width - 160 }}>
```

Comment:
you may be able to make `optionRowRight` set `width: 160` or similar and avoid using window dimensions

Reply:
tried some variants but they all make it bleed offscreen, or it ends up breaking the underlay, maybe I'm missing something obvious

![Simulator Screenshot - iPhone 14 Pro - 2024-04-04 at 12 13 03](https://github.com/daimo-eth/daimo/assets/6984346/8f5d53d4-e1e8-4496-a0a4-1d8a06b9a9d7)